src/machine.rs

```diff
@@ -140,8 +140,9 @@ pub enum Tag {
 
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
 static_assert_size!(Pointer<Tag>, 24);
+// FIXME: this would with in 24bytes but layout optimizations are not smart enough
 // #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-// static_assert_size!(Pointer<Option<Tag>>, 24);
+//static_assert_size!(Pointer<Option<Tag>>, 24);
```

Comment:
I am quite bummed by this... `Tag` now has two `NonZeroU64`, so we could use one to indicate `Wildcard` and one to indicate `None`, but it seems rustc is not smart enough for that. :(

I wonder if it's worth implementing the packing by hand, or if having the extra 8 bytes is just not such a big deal...